sw/banshee/src/runtime/common.rs

```diff
@@ -81,3 +83,23 @@ pub struct DmaState {
     size: u32,
     done_id: u32,
 }
+
+/// Store IRQ relevant CSRs
+#[derive(Default)]
+#[repr(C)]
+pub struct IrqState {
+    // a counter to check the interrupt state every sample_ctr instructions
+    pub sample_ctr: u32,
+    // interrupt control bits of the mstatus CSR
+    pub mstatus: u32,
+    // Machine interrupt enabled (same bits as mie csr)
+    pub mie: u32,
+    // machine interrupt pending (same bits as mip csr)
+    pub mip: u32,
+    // machine trap vector
+    pub mtvec: u32,
+    // machine exception PC
+    pub mepc: u32,
+    // machine cause
+    pub mcause: u32,
+}
```

Comment:
The only real comment I would have is that this seems a bit ad-hoc to me, I can see how `mcause` relates to interrupts but I would probably prefer if we can group the CSRs in something like a privileged state (in case we are extending the capabilities later). But I am okay with leaving it like this now.

Reply:
Agree, we could simply group all CSRs into a HashMap for example. @SamuelRiedel any suggetions?

Reply:
A hash map sounds like a good idea. But some special registers like `minstret`, which is accessed each instruction, might better be kept directly accessible.